src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,732 @@
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject, $rootType);
+            }
+            if ($schemaObject instanceof Reference && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromReference($schemaObject, $rootType);
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if ($schemaObject->items instanceof Schema && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject->items, $rootType);
+            }
+
+            if ($schemaObject->items instanceof Reference && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromReference($schemaObject->items, $rootType);
+            }
+        }
+
+        if (current($rootType) === true) {
+            return $this->formatPropertiesToArrayOfClassInstance($this->getClassNameFromReference($schema));
+        }
+
+        return $this->formatResponseProperties($this->getPropertiesFromReference($schema));
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return array
+     */
+    protected function formatPropertiesToArrayOfClassInstance(string $className): array
+    {
+        $refClass = str_replace('Attributes', '', $className);
+
+        return [$this->inflector->pluralize($refClass) => $refClass . '[]:' . $this->inflector->camelize($refClass)];
+    }
+
+    /**
+     * @param iterable $properties
+     *
+     * @return array
+     */
+    protected function formatResponseProperties(iterable $properties): array
+    {
+        $response = [];
+        foreach ($properties as $key => $schemaObject) {
+            if (isset($schemaObject->type) && $schemaObject->type !== 'array') {
+                $response[$key] = $schemaObject->type;
+
+                continue;
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if (isset($schemaObject->items->type)) {
+                $response[$key] = 'array[]:' . $schemaObject->items->type;
+
+                continue;
+            }
+
+            if (isset($schemaObject->items->properties) && isset($schemaObject->items->properties['type'])) {
+                $response[$key] = 'array[]:' . $schemaObject->items->properties['type']->type;
+
+                continue;
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return string
+     */
+    protected function getClassNameFromSchema(Schema $schema): string
```

Comment:
Which class name? I guess it should be 
```suggestion
    protected function getTransferNameFromSchema(Schema $schema): string
```